design/dsl/endpoint_test.go

```diff
@@ -3,114 +3,134 @@ package dsl_test
 import (
 	"testing"
 
+	"reflect"
+
 	"github.com/goadesign/goa/design"
 	. "github.com/goadesign/goa/design/dsl"
 	"github.com/goadesign/goa/eval"
 )
 
 func TestEndpoint(t *testing.T) {
 	cases := map[string]struct {
-		Expr        eval.Expression
-		Invocations int
-		Description string
-		Doc         string
-		DocURL      string
-		Name        string
-		RequestAttr map[string]design.Primitive
+		Expr   eval.Expression
+		DSL    func()
+		Assert map[string]func(t *testing.T, s *design.EndpointExpr)
```

Comment:
Why do we need  map here? looks like the tests below is calling helpers in one Assert function so with that model the map doesn't seem necessary?